src/clr-angular/progress/progress-bars/progress-bar.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component, Input, HostBinding } from '@angular/core';
+
+@Component({
+  selector: 'clr-progress-bar',
+  template: `
+  <ng-container>
+    <progress [id]="id" [attr.max]="max" [attr.value]="value" [attr.data-displayval]="displayValue"></progress>
+    <span *ngIf="displayAriaLive()" [attr.aria-live]="liveRegion">{{ displayValue }}</span>
+  </ng-container>
+  `,
+  host: {
+    '[attr.class]': 'elementClass',
```

Comment:
I'm not a fan of this pattern, because at first I thought you might be resetting any classes. I would rather add the targeted specific classes as necessary instead of trying to manage the class list ourselves. 

In fact, I would prefer to have the variations provided as attributes instead.

```
<clr-progress-bar danger fade></clr-progress-bar>
```

Then internally we build the necessary classes. That really internalizes the classes and structure, and I think aligns to future work.